index.html

```diff
@@ -2627,56 +2334,43 @@ <h3 id="applying">
           <code title="">categories</code> member
         </h3>
         <p>
-          The <dfn id="member-categories"><code>categories</code> member</dfn>
-          describes the expected application categories to which the web
-          application belongs.
+          The <dfn>categories</dfn> member describes the expected application
+          categories to which the web application belongs.
         </p>
         <p>
-          The <code>categories</code> member is only meant as a hint to
-          catalogs or stores listing web applications and it is expected that
-          these will make a best effort to find appropriate categories (or
-          category) under which to list the web application. Like search
-          engines and meta keywords, catalogs and stores are not required to
-          honor this hint.
+          The <a>categories</a> member is only meant as a hint to catalogs or
+          stores listing web applications and it is expected that these will
+          make a best effort to find appropriate categories (or category) under
+          which to list the web application. Like search engines and meta
+          keywords, catalogs and stores are not required to honor this hint.
         </p>
         <p>
-          The <dfn>steps for processing the <code>categories</code>
+          The <dfn>steps for post-processing the <code>categories</code>
           member</dfn> are given by the following algorithm. The algorithm
-          takes a <var>manifest</var> as an argument. This algorithm returns an
-          array of strings or <code>undefined</code>.
+          takes a <a>WebAppManifest</a> <var>manifest</var> as an argument.
+          This algorithm returns an <code>Array&lt;USVString&gt;</code>.
         </p>
         <ol>
-          <li>Let <var>categories</var> be an empty list.
+          <li>Let <var>A</var> be <var>manifest["categories"]</var>.
           </li>
-          <li>Let <var>unprocessed categories</var> be the result of calling
-          the <a>[[\GetOwnProperty]]</a> internal method of <var>manifest</var>
-          with argument "<code>categories</code>".
+          <li>Let <var>n</var> be the length of A.
           </li>
-          <li>If <var>unprocessed categories</var>is an array, then:
-            <ol>
-              <li>For each <var>potential category</var> in the array:
-                <ol>
-                  <li>
-                    <a>Trim</a>(<var>value</var>) and convert <a>to ASCII
-                    lowercase</a>
-                  </li>
-                  <li>Append <var>potential category</var> to
-                  <var>categories</var>
-                  </li>
-                </ol>
-              </li>
-            </ol>
+          <li>Initialize <var>i</var> to be 0.
```

Comment:
Does this have to be an index-based loop? Can you just write "For each element of *A*, *e*: Trim(*e*) and convert to ASCII lowercase"? (Optionally, "updating the value in *A*" if you are worried about someone taking it too literally.)

Reply:
Nope. Actually WebIDL uses it this way and I was looking for examples on how to write it properly without doing indices. @tobie is there a preferred wording?

Reply:
As a general rule, don't copy WebIDL's style... it's a very old spec and exhibits a ton of really bad practices (for legacy reasons).

Reply:
fixed